src/main/drivers/light_ws2811strip_hal.c

```diff
@@ -152,18 +149,14 @@ void ws2811LedStripDMAEnable(void)
         return;
     }
 
-    if (timerNChannel) {
-        if (HAL_TIMEx_PWMN_Start_DMA(&TimHandle, timerChannel, ledStripDMABuffer, WS2811_DMA_BUFFER_SIZE) != HAL_OK) {
-            /* Starting PWM generation Error */
-            ws2811LedDataTransferInProgress = 0;
-            return;
-        }
-    } else {
-        if (HAL_TIM_PWM_Start_DMA(&TimHandle, timerChannel, ledStripDMABuffer, WS2811_DMA_BUFFER_SIZE) != HAL_OK) {
-            /* Starting PWM generation Error */
-            ws2811LedDataTransferInProgress = 0;
-            return;
-        }
+    if (DMA_SetCurrDataCounter(&TimHandle, timerChannel, ledStripDMABuffer, WS2811_DMA_BUFFER_SIZE) != HAL_OK) {
+        /* DMA set error */
+        ws2811LedDataTransferInProgress = 0;
+        return;
     }
+    /* Reset timer counter */
+    __HAL_TIM_SET_COUNTER(&TimHandle,0);
+    /* Enable channel DMA requests */
+    TIM_DMACmd(&TimHandle,timerChannel,ENABLE);
```

Comment:
This is problematic - it will probably trigger DMA transfer immediately (DMA request is already set). With DMA FIFO enabled, this may fail the transfer (no data in FIFO to satisfy peripheral request, see reference manual).
It will work now, but can cause big headache when someone tries to optimize it